src/transports/http.c

```diff
@@ -359,6 +360,9 @@ static int on_headers_complete(http_parser *parser)
 					t->parse_error = PARSE_ERROR_REPLAY;
 					return 0;
 				}
+			} else {
+				giterr_set(GITERR_NET, "authentication failed, incorrect credentials");
```

Comment:
This error message is either imprecise or incorrect.  We don't know if credentials are "correct" or "incorrect", we only know that we got an Unauthorized.  The credentials could be "correct" (authenticate a particular user) but that user doesn't have authorization to the resource.